camelot/view/controls/delegates/delegatemanager.py

```diff
@@ -68,15 +70,10 @@ def _commit_data(self, editor):
     def _close_editor(self, editor, hint):
         self.closeEditor.emit(editor, hint )
 
-    def paint(self, painter, option, index):
-        """Use a custom delegate paint method if it exists"""
-        delegate = self.get_column_delegate(index.column())
-        delegate.paint(painter, option, index)
-
     def createEditor(self, parent, option, index):
```

Comment:
Deze code doet de camelot unit testen nog wel falen.

Indien er nog geen delegate is, wordt dit wel opgevangen door een QWidget te returnen, maar hierdoor krijg je elders ook nog exceptions:

```
======================================================================
ERROR: test_form (test.test_view.FormTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/camelot/test/test_view.py", line 472, in test_form
    self.grab_widget(Movie.Admin.form_display.render(self.widgets))
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 433, in render
    widget = DelayedTabWidget( widgets, self.tabs, parent )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 308, in __init__
    self.render_tab( i )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 334, in render_tab
    tab_form_widget = tab_form.render( self._widgets, tab_widget, False )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 182, in render
    f = field.render( widgets, parent, False )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 474, in render
    f = form.render( widgets, widget, False )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/forms.py", line 216, in render
    label = widgets.create_label( field, editor, form_widget )
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/controls/formview.py", line 110, in create_label
    editor.set_label(widget_label)
AttributeError: 'QWidget' object has no attribute 'set_label'
```

Kunnen we die return `QtWidgets.QWidget( parent = parent )` nier vervangen door `return CustomEditor(parent=parent)`? Dan hebben we de methodes als `set_label` etc. Dan werken de unit testen ook terug.

Reply:
Ter info, de opgevangen exceptions:

```
camelot.view.controls.delegates.delegatemanager: ERROR: Programming Error : could not createEditor editor data for editor at column 0
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/controls/delegates/delegatemanager.py", line 80, in createEditor
    editor = delegate.createEditor(parent, option, index)
AttributeError: 'NoneType' object has no attribute 'createEditor'
camelot.view.controls.delegates.delegatemanager: DEBUG: setting editor data for column 0
camelot.view.controls.delegates.delegatemanager: ERROR: Programming Error : could not set editor data for editor at column 0
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/controls/delegates/delegatemanager.py", line 94, in setEditorData
    delegate.setEditorData(editor, index)
AttributeError: 'NoneType' object has no attribute 'setEditorData'
camelot.view.controls.delegates.delegatemanager: ERROR: value that could not be set : None
camelot.view.controls.delegates.delegatemanager: ERROR: editor that failed QWidget cover_editor
camelot.view.forms: DEBUG: rendering Form
camelot.view.controls.delegates.delegatemanager: ERROR: Programming Error : could not createEditor editor data for editor at column 1
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/controls/delegates/delegatemanager.py", line 80, in createEditor
    editor = delegate.createEditor(parent, option, index)
AttributeError: 'NoneType' object has no attribute 'createEditor'
camelot.view.controls.delegates.delegatemanager: DEBUG: setting editor data for column 1
camelot.view.controls.delegates.delegatemanager: ERROR: Programming Error : could not set editor data for editor at column 1
Traceback (most recent call last):
  File "/home/timvdm/vortex-blue-launcher/camelot/camelot/view/controls/delegates/delegatemanager.py", line 94, in setEditorData
    delegate.setEditorData(editor, index)
AttributeError: 'NoneType' object has no attribute 'setEditorData'
camelot.view.controls.delegates.delegatemanager: ERROR: value that could not be set : None
```

Reply:
dit was me dus nt opgevallen, cfr volgende pr, die zou eigenlijk ook beter screenshots moeten geven, namelijk met available data.